cypress/e2e/tests/pages/fleet/gitrepo.spec.ts

```diff
@@ -94,33 +96,102 @@ describe('Git Repo', { tags: ['@fleet', '@adminUser'] }, () => {
+
+          cy.intercept('PUT', 'v1/userpreferences/*').as(`prefUpdateEnUs`);
+          prefPage.languageDropdownMenu().clickOption(1);
+          cy.wait('@prefUpdateEnUs').then(({ response }) => {
+            expect(response?.statusCode).to.eq(200);
+            expect(response?.body.data).to.have.property('locale', 'en-us');
+          });
+          prefPage.languageDropdownMenu().isClosed();
         })
       ;
     });
 
-    after(() => {
-      const prefPage = new PreferencesPagePo();
-
-      // https://github.com/rancher/dashboard/issues/9984 reset lang to EN so that delete action can be performed
-      prefPage.goTo();
-      prefPage.languageDropdownMenu().checkVisible();
-      prefPage.languageDropdownMenu().toggle();
-      prefPage.languageDropdownMenu().isOpened();
-
-      cy.intercept('PUT', 'v1/userpreferences/*').as(`prefUpdateEnUs`);
-      prefPage.languageDropdownMenu().clickOption(1);
-      cy.wait('@prefUpdateEnUs').then(({ response }) => {
-        expect(response?.statusCode).to.eq(200);
-        expect(response?.body.data).to.have.property('locale', 'en-us');
-      });
-      prefPage.languageDropdownMenu().isClosed();
+    // testing https://github.com/rancher/dashboard/issues/9866
+    it('in git repo details view we should display the correct bundle count', () => {
+      const listPage = new FleetGitRepoListPagePo();
+
+      const basicRepos = [
+        {
+          name:   'e2e-git-repo1-test-bundle-count',
+          repo:   'https://github.com/rancher/fleet-examples.git',
+          branch: 'master',
+          path:   'simple'
+        },
+        {
+          name:   'e2e-git-repo2-test-bundle-count',
+          repo:   'https://github.com/rancher/fleet-examples.git',
+          branch: 'master',
+          path:   'single-cluster/helm'
+        }
+      ];
+
+      // create first git-repo in fleet-local
+      gitRepoCreatePage.goTo();
+      gitRepoCreatePage.waitForPage();
+
+      gitRepoCreatePage.selectWorkspace('fleet-local');
+
+      gitRepoCreatePage.setRepoName(basicRepos[0].name);
+      gitRepoCreatePage.setGitRepoUrl(basicRepos[0].repo);
+      gitRepoCreatePage.setBranchName(basicRepos[0].branch);
+      gitRepoCreatePage.gitRepoPaths().setValueAtIndex(basicRepos[0].path, 0);
+      gitRepoCreatePage.goToNext();
+      gitRepoCreatePage.create();
 
+      // create second git-repo in fleet-local
+      listPage.waitForPage();
+      listPage.repoList().create();
+
+      gitRepoCreatePage.selectWorkspace('fleet-local');
+
+      gitRepoCreatePage.setRepoName(basicRepos[1].name);
+      gitRepoCreatePage.setGitRepoUrl(basicRepos[1].repo);
+      gitRepoCreatePage.setBranchName(basicRepos[1].branch);
+      gitRepoCreatePage.gitRepoPaths().setValueAtIndex(basicRepos[1].path, 0);
+      gitRepoCreatePage.goToNext();
+      gitRepoCreatePage.create();
+
+      listPage.waitForPage();
+      listPage.selectWorkspace('fleet-local');
+
+      listPage.goToDetailsPage(basicRepos[1].name);
```

Comment:
The issue's steps linked in the PR describes to go navigate back to the first gitRepo. Or is it that any gitrepo will behave the same without the fix.

Reply:
I can answer this since I filed that issue: the bug was reproducible with any repos whose name shared the same prefix followed by a hyphen. Since they both start with `e2e-`, the `e2e-git-repo1-test-bundle-count` `e2e-git-repo2-test-bundle-count`  detail views would have each showed the bundle count from both (2) instead of the correct count (1). So this should be okay.

Reply:
Ah makes sense thanks @mantis-toboggan-md